core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the skill-selector component.
+ */
+
+import cloneDeep from 'lodash/cloneDeep';
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { CategorizedSkills } from 'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { SkillSummary } from 'core/templates/domain/skill/skill-summary.model';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { GroupedSkillSummaries } from 'pages/skill-editor-page/services/skill-editor-state.service';
+import { FilterForMatchingSubstringPipe } from 'filters/string-utility-filters/filter-for-matching-substring.pipe';
+import { ShortSkillSummary } from 'core/templates/domain/skill/ShortSkillSummaryObjectFactory';
+
+require('domain/utilities/url-interpolation.service.ts');
+
+interface SubTopicFilterDict {
+  [topicName: string] : { subTopicName: string; checked: boolean }[]
+}
+
+@Component({
+  selector: 'skill-selector',
+  templateUrl: './skill-selector.component.html',
+})
+export class SkillSelectorComponent implements OnInit {
+  // If countOfSkillsToPrioritize > 0, then sortedSkillSummaries should
+  // have the initial 'countOfSkillsToPrioritize' entries of skills with
+  // the same priority.
+  @Input() sortedSkillSummaries: GroupedSkillSummaries;
+  @Input() selectedSkillId: string;
+  @Input() countOfSkillsToPrioritize: number;
+  @Input() categorizedSkills: CategorizedSkills;
+  @Input() untriagedSkillSummaries: SkillSummary[];
+  @Input() allowSkillsFromOtherTopics: boolean;
+  @Output() selectedSkillIdChanged: EventEmitter<string> = new EventEmitter();
+  currCategorizedSkills: CategorizedSkills = null;
+  selectedSkill: string = null;
+  skillFilterText: string = '';
+  topicFilterList: { topicName: string ; checked: boolean }[] = [];
+  subTopicFilterDict: SubTopicFilterDict = {};
+  intialSubTopicFilterDict: SubTopicFilterDict = {};
+
+  private filterForMatchingSubtringPipe: FilterForMatchingSubstringPipe = (
+    new FilterForMatchingSubstringPipe()
+  );
+
+  constructor() {}
```

Comment:
If you are using `@Injectable()` decorator for FilterForMatchingSubstringPipe in the `filter-for-matching-substring.pipe.ts` then you could just remove this and add it to the constructor.
```suggestion
  constructor(
    private filterForMatchingSubtringPipe: FilterForMatchingSubstringPipe) {}
```

Reply:
Done